GooglePay/src/main/java/com/braintreepayments/api/googlepay/GooglePayClient.kt

```diff
@@ -0,0 +1,620 @@
+                    JSONArray().put(
+                        JSONObject().put("type", "CARD")
+                            .put(
+                                "parameters", JSONObject().put(
+                                    "allowedAuthMethods",
+                                    JSONArray().put("PAN_ONLY").put("CRYPTOGRAM_3DS")
+                                )
+                                    .put("allowedCardNetworks", allowedCardNetworks)
+                            )
+                    )
+                )
+
+                if (request != null) {
+                    json.put(
+                        "existingPaymentMethodRequired",
+                        request.isExistingPaymentMethodRequired
+                    )
+                }
+            } catch (ignored: JSONException) {
+            }
+            val readyToPayRequest = IsReadyToPayRequest.fromJson(json.toString())
+            internalGooglePayClient.isReadyToPay(context, configuration, readyToPayRequest, callback)
+        }
+    }
+
+    /**
+     * Get Braintree specific tokenization parameters for a Google Pay. Useful for when full control
+     * over the [PaymentDataRequest] is required.
+     *
+     *
+     * [PaymentMethodTokenizationParameters] should be supplied to the
+     * [PaymentDataRequest] via
+     * [ ][PaymentDataRequest.Builder.setPaymentMethodTokenizationParameters]
+     * and [&lt;Integer&gt;][Collection] allowedCardNetworks should be supplied to the
+     * [CardRequirements] via
+     * [CardRequirements.Builder.addAllowedCardNetworks]}.
+     *
+     * @param callback [GooglePayGetTokenizationParametersCallback]
+     */
+    fun getTokenizationParameters(
+        callback: GooglePayGetTokenizationParametersCallback
+    ) {
+        braintreeClient.getConfiguration { configuration: Configuration?, e: Exception? ->
+
+            if (configuration != null) {
+                callback.onTokenizationParametersResult(
+                    GooglePayTokenizationParameters.Success(
+                        getTokenizationParameters(configuration, braintreeClient.authorization),
+                        getAllowedCardNetworks(configuration)
+                    )
+                )
+            } else {
+                if (e != null) {
+                    callback.onTokenizationParametersResult(GooglePayTokenizationParameters.Failure(e))
+                } else {
+                    callback.onTokenizationParametersResult(null)
+                }
+            }
+        }
+    }
+
+    /**
+     * Start the Google Pay payment flow. This will return [GooglePayPaymentAuthRequestParams] that are
+     * used to present Google Pay payment sheet in
+     * [GooglePayLauncher.launch]
+     *
+     * @param request  The [GooglePayRequest] containing options for the transaction.
+     * @param callback [GooglePayPaymentAuthRequestCallback]
+     */
+    @SuppressWarnings("LongMethod", "MaximumLineLength")
```

Comment:
Which line is causing the max line length error? I think we should be able to resolve that instead of suppress